### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by MMullis07**
Looking forward to week 2!!

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -7,60 +7,111 @@
...
+//         return `${temp} is below boiling point.`
+//     } else if (temp > 212) {
+//         return `${temp} is above boiling point.`
+//     } else {
+//         return `${temp} is at boiling point.`
+//     }
+// }
+// console.log(tempBP(temp1))
+// console.log(tempBP(temp2))
+// console.log(tempBP(temp3))
```
Comment (andeeisaacs) [inline code-challenges.js:25]: Excellent!

File: code-challenges.js [hunk 2]
```diff
@@ -7,60 +7,111 @@
...
+// // Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45]
 
+    // var myNumbers1 = [3, 7, 0, 6, -9]
+    // var newNumbers = []
+    // const mult5 = (array) => {
+    //     for(let i=0; i < array.length; i++) {
+    //        newNumbers.push(array[i] * 5)
+    //     }
+    //     return newNumbers
+    // }
```
Comment (andeeisaacs) [inline code-challenges.js:39]: Nice and clean. Love it!  Good use of .push.

File: code-challenges.js [hunk 3]
```diff
@@ -7,60 +7,111 @@
...
 
-var myNumbers2 = [8, -7, 0, 6, 2]
-
-
-
+//  var myNumbers2 = [8, -7, 0, 6, 2]
+//  const multi5 = myNumbers2.map(value => {
+//      return value * 5
+//  })
+//  console.log(multi5)
```
Comment (andeeisaacs) [inline code-challenges.js:51]: As it is right now, this function is not reusable on anything other than myNymbers2, and cannot be used with any other variable. We need to remember the reason we create functions is for them to be dynamic and reusable pieces of code.

File: code-challenges.js [hunk 4]
```diff
@@ -7,60 +7,111 @@
...
+//    var stringWithVowels1 = "HeyThereLearnStudent"
+//    var stringWithVowels2 = "ILoveJavaScript"
+   
+//    const noVowels = (string) => {
+//        var newArr = string.split("")
+//        return newArr.filter(value => {
+//            return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u"}).join("")
+//    }
+//    console.log(noVowels(stringWithVowels1))
+//    console.log(noVowels(stringWithVowels2))
```
Comment (andeeisaacs) [inline code-challenges.js:69]: Almost! We needed to also check for uppercase vowels as well. If we look at our expected output for `var stringWithVowels2 = "ILoveJavaScript"` 

Our expected outcome is `"LvJvScrpt"`

But what the function is returning is `ILvJvScrpt`, which is still including the uppercase i.

Take a look into .toLowerCase() and how you can include that in your filter when checking for the vowels!

File: code-challenges.js [hunk 5]
```diff
@@ -7,60 +7,111 @@
...
+// const noVowels = (string) => {
+//     if (typeof string === "string") {
+//         return string.split("").filter(value => {
+//             return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u"}).join("")
+//     } else {
+//         return `${string} is not a string.`
+//     }
+// }
+//     console.log(noVowels(notAString1))
+//     console.log(noVowels(notAString2))
```
Comment (andeeisaacs) [inline code-challenges.js:88]: Excellent! Love the use of chaining methods!

File: code-challenges.js [hunk 6]
```diff
@@ -7,60 +7,111 @@
...
-
-var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
-
-
-
+// const catsOnly = (array) => {
+//     return array.filter(value => {
+//         return value.animal === "cat"
+//     })
+// }
```
Comment (andeeisaacs) [inline code-challenges.js:100]: Nice and clean! We can clean it up even more, since we dont need to open up a block of code inside of our filter, since its just checking for one thing- 

`return array.filter(value => value.animal === "cat ) `

Again, nothing wrong with how you did it! Just a way to keep it more concise :)

File: code-challenges.js [hunk 7]
```diff
@@ -7,60 +7,111 @@
...
+
+// const nonCats = (array) => {
+//     let nonKitties = array.filter(value => {
+//         return value.animal !== "cat"
+//     })
+//     return nonKitties.map(value => {
+//         return value.name
+//     })
+// }
+// console.log(nonCats(toonimals))
```
Comment (andeeisaacs) [inline code-challenges.js:117]: Beautiful! 👍 

Another way of doing this I'd like you to look at is: 

`const nonCats = (array) => {`
` return array.filter(value => value.animal !== "cat").map(value => value.name)`
`}`

This is the exact same thing, but not needing to create a seperate variable to hold the non cats or expanding out the code block for each filter and map. We can just chain the map onto the filter function end, as it works in the order of operation! 

Once again, how you did it is TOTALLY excellent, but I just wanted to give you something else to consider!

File: interview-questions.md [hunk 8]
```diff
@@ -7,64 +7,98 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: 
+  A function is a callable block of code used to compute data by passing it arguments. It can be used to peform basic mathematics to filtering through arrays.
```
Comment (andeeisaacs) [inline interview-questions.md:11]: True! Additionally, I'd want to expand on WHY we would use one, as the interviewer is asking. This would be a good place to add that its useful for creating dynamic and reusable blocks of code to manipulate data passed into it.

File: interview-questions.md [hunk 9]
```diff
@@ -7,64 +7,98 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  A function takes some input and returns an output and there should be some sort of relationship between the input and the output. To use it you must define in the scope from which you wish to call it. You can pass through primitive parameters such as numbers or objects such as arrays. You must define the function and then call it. 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: 
+    .filter() goes through an array and filters out anything that matches the data type you are looking for and produces a new array. 
+  
+    .map() goes through an array and returns an array of the same length.
```
Comment (andeeisaacs) [inline interview-questions.md:22]: Totally! Good answer!

File: interview-questions.md [hunk 10]
```diff
@@ -7,64 +7,98 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+    .map() Iterates through an array and returns a new array of same length. Like .filter() it also takes in three arguments (value, index, and array). It will call a function for every array element in order of the original array. It does not change the original array.
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer:  
+    console.log() will actually cause whatever it is we are calling to be printed out in the console. 
+                
+    return simply stores whatever information we are "returning" to that codeblock/function/etc.
```
Comment (andeeisaacs) [inline interview-questions.md:35]: 👍

File: interview-questions.md [hunk 11]
```diff
@@ -7,64 +7,98 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+    The return state stops the execution of a function and returns a value from inside that function. It requires a value if one is not provided it will return "undefined". It also must be used inside a function.
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: 
+    The argument is what the function takes in in order for it to run
```
Comment (andeeisaacs) [inline interview-questions.md:46]: I would want to expand a bit more on this- the argument is a placeholder for  data that is passed into the function at the time of call to be manipulated.

File: interview-questions.md [hunk 12]
```diff
@@ -7,64 +7,98 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
+    Techniques: Usually done in pairs. One person is driver, other is navigator. They set a timer or complete a set number of tasks in their current roles. Once the timer goes off or they complete that number of tasks you switch roles. Continue this cycle until finish!
 
-  Researched answer:
+    Driver- Focuses on typing and listens to the navigator telling him what     to type
 
+    Navigator- Focuses on the task that needs solved/completed and instructs the driver on what to type up
```
Comment (andeeisaacs) [inline interview-questions.md:59]: Totally! Great answers!

File: interview-questions.md [hunk 13]
```diff
@@ -7,64 +7,98 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+  Researched answer:
+    Two developers work using only one machine. They may each have their own keyboard and mouse. One programmer acts as the driver while the other will proofread, spell check, and figure out what to code next. The roles are then switched. Studies have shown that paired programming results in 15% less bugs in code. 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: 
+    Test-driven development. Describe the test. Set the Test. Write out test expectations. Test to see that it will fail appropriately. Finally, write code that makes test pass!
```
Comment (andeeisaacs) [inline interview-questions.md:69]: Absolutely! I'd also want to add in the name of the process, called 'Red-Green-Refactor'.

File: testing-with-jest.test.js [hunk 14]
```diff
@@ -8,20 +8,37 @@
...
-
+//Describe the test
+describe ("addThemUp", () => {
+    //Set the test
+    test("addThemUp that takes two numbers as arguments and returns the sum.", () => {
+        //Write expectations
+        expect(addThemUp(4, 28)).toEqual(32)
+        expect(addThemUp(5, 95)).toEqual(100)
+    })
+})
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:19]: Excellent test. I like that you added more than just one expect!

File: testing-with-jest.test.js [hunk 15]
```diff
@@ -8,20 +8,37 @@
...
-
+//Describe the test
+describe ("triangleArea", () => {
+    //Set the test
+    test("riangleArea that takes the base and the height of a triange and returns the area. The area of a triange is the base times the height divided by two.", () => {
+        //Write expectations
+        expect(triangleArea(4, 4)).toEqual(8)
+        expect(triangleArea(15, 4)).toEqual(30)
+    })
+})
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:39]: Perfect!

**Review by andeeisaacs — APPROVED** [review]
Matt! Excellent work on your first week assessment! I would be super happy if I were you!

A few notes: 
- You did not utilize pseudo coding at all in any of your challenges. Pseudo coding is beneficial not only for helping you plan out/visualize your code, but also for notes to yourself later. Besides those two things, it's also important in communication, as that is what code truly is. It's not only communication between you and the computer, but communication to other humans. Without commenting/pseudo coding, it's hard to follow your logic flow and steps you took in-order to find the solutions to your challenges. Take time to pseudo code each line of your code! It will help a whole lot towards gaining these habits that are required by companies in the industry. 

- Your functions were FANTASTIC. I loved them! The only things I would encourage you to do is really pay attention to your expected outputs to make sure they match up, including edge cases. 

Overall, you did fantastic and I look forward to seeing where you are on your week 2! 

Feel free to merge!

**Comment by MMullis07** [comment]
Week 1 D-U-N DONE!
